extension/observer/ecsobserver/task.go

```diff
@@ -65,3 +80,67 @@ func (t *Task) ContainerLabels(containerIndex int) map[string]string {
 	}
 	return labels
 }
+
+// PrivateIP returns private ip address based on network mode.
+// EC2 launch type can use hsot/brige mode and the private ip is the EC2 instance's ip.
+// awsvpc has its own ip regardless of launch type.
+func (t *Task) PrivateIP() (string, error) {
+	arn := aws.StringValue(t.Task.TaskArn)
+	switch aws.StringValue(t.Definition.NetworkMode) {
+	// Default network mode is bridge on EC2
+	case "", ecs.NetworkModeHost, ecs.NetworkModeBridge:
+		if t.EC2 == nil {
+			return "", fmt.Errorf("task has no network mode and no ec2 info %s", arn)
```

Comment:
No, they should be in same case, when using ec2, as long as you are not using awspc, the task ip is host ip. Though I might need to double check fargate behaviour, fargate always uses awsvpc but I am not sure if the API accepts `""` as network mode (it does for ec2).

Reply:
Checked fargate, it has to be awsvpc https://docs.aws.amazon.com/AmazonECS/latest/userguide/fargate-task-defs.html#fargate-tasks-networkmode. So `""` means it must be ECS EC2 bridge, otherwise it will get rejected by ECS API and we won't see it the extension.